### PR TITLE
Allows to fetch new sounds for annotation

### PR DESCRIPTION
**Pull request by aframires**
Users are now able to get new sounds to annotate which are selected according to their importance for the dataset. The importance metric uses the number of annotations for each loop, author, pack and genres, to maximise the distribution of these.

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -48,7 +51,8 @@ def assign_more_sounds_to_user(username, N=default_N_assign_more_sounds):
     # TODO: this function randomly assigns a maximum of N new annotations
     # we should do a real implementation which choses wisely which sounds
     # to assign
-    new_ids = random.sample(all_sound_ids, N)
+    new_ids = select_relevant_sounds(PATH_TO_ANNOTATIONS, joined_metadata, genres_file, all_sound_ids, N)
+    print(new_ids)
```
Comment (ffont) [inline app.py]: remove this print please :)

File: app.py [hunk 2]
```diff
@@ -24,9 +26,10 @@
 all_sound_ids = json.load(open(PATH_TO_ALL_SOUND_IDS, 'rb')) 
 sound_id_user = json.load(open(PATH_TO_SOUND_IDS_PER_USER,'rb'))
 genres_file = json.load(open(PATH_TO_GENRE_FILE, 'rb'))
+joined_metadata = json.load(open(PATH_TO_JOINED_METADATA, 'rb'))
 
-default_N_assign_more_sounds = 5
-enable_auto_assign_annotations = False
+default_N_assign_more_sounds = 10
```
Comment (ffont) [inline app.py:31]: maybe let's set it to 25?

File: getsounds.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+#PATH_TO_ALL_SOUND_IDS = os.path.join(PATH_TO_FSL10K, 'metadata_sound_ids_list.json')
+#PATH_TO_METADATA = os.path.join(PATH_TO_FSL10K, "metadata.json")
+#PATH_TO_GENRES = os.path.join(PATH_TO_FSL10K, "parent_genres.json")
+#PATH_TO_ANNOTATIONS =  os.path.join(PATH_TO_FSL10K, 'annotations/')
+default_N_assign_more_sounds = 10
+
+#metadata = json.load(open(PATH_TO_METADATA, 'rb'))
+#genres_metadata = json.load(open(PATH_TO_GENRES, 'rb'))
+#all_sound_ids = json.load(open(PATH_TO_ALL_SOUND_IDS,'rb'))
```
Comment (ffont) [inline getsounds.py]: Remove all these commented lines plz

File: getsounds.py [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+def select_relevant_sounds(annotations_path, metadata, genre_metadata, all_sound_ids, N=default_N_assign_more_sounds):
+
+    sounds_annotated = compile_annotated_sounds(annotations_path)
+
+    anno_weight=1000
+    auth_weight=1
+    pack_weight=1
+    genre_weight=-10
+
```
Comment (ffont) [inline getsounds.py:102]: You can add a comment before each of the weight definition to explain what does it mean and why we set it that way

File: getsounds.py [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+        if pack_sounds.get(metadata[sound]["pack"]) != None:
+            num_pack = pack_sounds[metadata[sound]["pack"]]
+        
+        gen_importance = genre_importance(genre_metadata.get(sound,[]), genre_sounds)
+        irrelevance = num_annotated*anno_weight + num_author*auth_weight + num_pack*pack_weight + gen_importance*genre_weight
+        sound_irrelevance_list.append((sound,irrelevance))
+    
+    sound_irrelevance_sorted = sorted(sound_irrelevance_list, key=lambda x: x[1])
+    sound_irrelevance_ids = [lis[0] for lis in sound_irrelevance_sorted]
+    print(sound_irrelevance_ids[0:N])
```
Comment (ffont) [inline getsounds.py]: remove the print

**Comment by ffont** [comment]
Added a couple of comments. After the comments are addressed I think this will be ready to merge. Next step will be some sort of "login with google" (or similar) so that we don't have to prepare the `users.json` file with all potential users. 

Another minor comment is that the PR description is not very good. If you want to give details you can explain how the algorithm works. If you don't want to give details you can just say what it does (which is fine). The list of IDs to "check" algorithm output however is not vey helpful because it depends on local data (so might give different results if you have a different `metadata` file) and there are no "instructions" for how to get the list of IDs and check with the ones in the description. Personally I'd just say what the algorithm does for this PR.

File: app.py (hunk 2 above)
Comment (aframires) [reply app.py:31]: I prefer to set it for a low number that users can annotate just after asking for more. If we take 25 sounds and don't finish annotating them, the system will not know they have been attributed to someone already and, if anyone asks for more sounds in the meantime, there's the possibility of these sounds to be assigned twice.

**Comment by aframires** [comment]
The code is now addressing @ffont reviews.

**Pull request merged**